pkg/nodeagent/files/files.go

```diff
@@ -89,7 +89,7 @@ func copyAndSync(fs afero.Afero, source, destination string) error {
 		return fmt.Errorf("destination directory %q could not be created", path.Dir(destination))
 	}
 
-	dstFile, err := fs.OpenFile(destination, os.O_CREATE|os.O_RDWR, 0755)
+	dstFile, err := fs.OpenFile(destination, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0700)
```

Comment:
For a quick fix, I'd suggest to use for now
```
dstFile, err := fs.OpenFile(destination, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0600)
```

That should be good enough for the `Copy` (shouldn't make a difference) and `moveCrossDevice` (still works if `tmpFilePath` already exists) cases for now.

Reply:
There is always room for improvement 😄 
I was afk today, but address your feedback in a separate PR.